src/PulseAudioManager.vala

```diff
@@ -200,12 +316,23 @@ public class Sound.PulseAudioManager : GLib.Object {
                         break;
 
                     case PulseAudio.Context.SubscriptionEventType.REMOVE:
-                        var device = output_devices.get (index);
-                        if (device != null) {
-                            device.removed ();
-                            output_devices.unset (index);
+                        debug ("subscribe_callback:SINK:REMOVE");
+                        foreach (var device in output_devices.values) {
+                            if (device.sink_index == index) {
+                                debug ("  updating device: %s", device.id);
```

Comment:
I think that if you want to provide some sort of formatting to debug output you should use `\t` instead of just spaces. This applies to every debug call.